BrowserStack.java

```diff
@@ -13,19 +15,25 @@
  *         printStack() - prints the Stack preserving all its values
  *         size()       - returns the length of the stack
  *         isEmpty()    - return true if stack is not empty else false
- *
+ *o--
  */
 public class BrowserStack {
 
     String[] myStack;
     int top;
-    int STACK_SIZE = 16; //INITIAL STACK SIZE, GROW IF ITS NEEDED
+    int STACK_SIZE = 2; //INITIAL STACK SIZE, GROW IF ITS NEEDED
 
     //Constructor to initialize your stack
     BrowserStack(){
         this.myStack = new String[STACK_SIZE];
         this.top = -1;
     }
+    private void createNewStack(){
+    	String[] newStack = new String[STACK_SIZE * 2];
+    	newStack = Arrays.copyOf(myStack, newStack.length);
+    	STACK_SIZE = STACK_SIZE *2;
+    	myStack = newStack;
```

Comment:
Old myStack is thrown away and so all the old elements in it. In this program it is ok.